logstash-core/src/test/java/org/logstash/ackedqueue/QueueTest.java

```diff
@@ -390,4 +390,121 @@ public void reachMaxUnreadWithAcking() throws IOException, InterruptedException,
         assertThat(q.unreadCount, is(equalTo(1L)));
     }
 
+    @Test(timeout = 5000)
+    public void reachMaxSizeTest() throws IOException, InterruptedException, ExecutionException {
+        Queueable element = new StringElement("0123456789"); // 10 bytes
+
+        int singleElementCapacity = ByteBufferPageIO.HEADER_SIZE + ByteBufferPageIO._persistedByteCount(element.serialize().length);
+
+        // allow 10 elements per page but only 100 events in total
+        Settings settings = TestSettings.getSettings(singleElementCapacity * 10, singleElementCapacity * 100);
+
+        TestQueue q = new TestQueue(settings);
+        q.open();
+
+        int ELEMENT_COUNT = 90; // should be able to write 99 events before getting full
+        for (int i = 0; i < ELEMENT_COUNT; i++) {
+            long seqNum = q.write(element);
+        }
+
+        assertThat(q.isFull(), is(false));
+
+        // we expect this next write call to block so let's wrap it in a Future
+        Callable<Long> write = () -> {
+            return q.write(element);
+        };
+
+        ExecutorService executor = Executors.newFixedThreadPool(1);
+        Future<Long> future = executor.submit(write);
+
+        while (!q.isFull()) { Thread.sleep(1); }
```

Comment:
Java `Thread.sleep()` accepts a milliseconds parameter, maybe `1` is a bit too small? I suggest `10`?  I know you used the same value as the similar test I wrote but I think I might have been too aggressive :P